dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/ContainerFactoryImpl.java

```diff
@@ -891,19 +904,36 @@ private void buildFindContainersQuery(final ContainerAPI.SearchParams searchPara
 		}
 
 		if(UtilMethods.isSet(searchParams.siteId())) {
-			query.append(" and identifier.host_inode = '");
-			query.append(searchParams.siteId()).append('\'');
+			// Use parameterized query to prevent SQL injection
+			query.append(" and identifier.host_inode = ?");
+			// Validate that siteId is a valid UUID
+			final String siteId = searchParams.siteId();
+			if (!UUIDUtil.isUUID(siteId)) {
+				throw new DotSecurityException("Invalid site ID format: " + siteId);
+			}
+			paramValues.add(siteId);
 		}
 
 		if(UtilMethods.isSet(searchParams.containerInode())) {
-			query.append(" and asset.inode = '");
-			query.append(searchParams.containerInode()).append('\'');
+			// Use parameterized query to prevent SQL injection
+			query.append(" and asset.inode = ?");
+			// Validate that containerInode is a valid UUID
+			final String containerInode = searchParams.containerInode();
+			if (!UUIDUtil.isUUID(containerInode)) {
+				throw new DotSecurityException("Invalid container inode format: " + containerInode);

```

Comment:
same as above

Reply:
perhaps a new Utiltiy method that validates if we're receiving a UUID and a valid internal identifier like SYSTEM_HOST or even SYSTEM_FOLDER

Reply:
FIXED: Now uses isValidIdentifier(containerInode) with same flexible validation